src/hooks-worker-client.coffee

```diff
@@ -174,35 +147,36 @@ class HooksWorkerClient
 
   spawnHandler: (callback) ->
     pathGlobs = [].concat @runner.hooks?.configuration?.options?.hookfiles
-    handlerCommandArgs = @handlerCommandArgs.concat pathGlobs
+    handlerCommandArgs = @handlerCommandArgs.concat(pathGlobs)
 
-    logger.info("Spawning `#{@language}` hooks handler process.")
-    @handler = crossSpawn.spawn @handlerCommand, handlerCommandArgs
+    logger.info("Spawning '#{@language}' hooks handler process.")
+    @handler = spawn(@handlerCommand, handlerCommandArgs)
 
-    @handler.stdout.on 'data', (data) ->
+    @handler.stdout.on('data', (data) ->
       logger.info("Hooks handler stdout:", data.toString())
-
-    @handler.stderr.on 'data', (data) ->
+    )
+    @handler.stderr.on('data', (data) ->
       logger.info("Hooks handler stderr:", data.toString())
+    )
 
-    @handler.on 'exit', (status) =>
-      if status?
-        if status isnt 0
-          msg = "Hooks handler process '#{@handlerCommand}' exited with status: #{status}"
-          logger.error(msg)
-          @runner.hookHandlerError = new Error(msg)
-      else
-        # No exit status code means the hook handler was killed
-        unless @handlerKilledIntentionally
-          msg = "Hooks handler process '#{@handlerCommand}' was killed."
-          logger.error(msg)
-          @runner.hookHandlerError = new Error(msg)
-      @handlerEnded = true
-
-    @handler.on 'error', (error) =>
-      @runner.hookHandlerError = error
-      @handlerEnded = true
+    @handler.on('signalTerm', ->
+      logger.verbose('Gracefully terminating the hooks handler process')
+    )
+    @handler.on('signalKill', ->
+      logger.verbose('Killing the hooks handler process')
+    )
 
+    @handler.on('crash', (exitStatus, killed) =>
+      if killed
+        msg = "Hooks handler process '#{@handlerCommand} #{handlerCommandArgs.join(' ')}' was killed."
+      else
+        msg = "Hooks handler process '#{@handlerCommand} #{handlerCommandArgs.join(' ')}' exited with status: #{exitStatus}"
+      logger.error(msg)
```

Comment:
Yup, it's non-standard termination of the hook handler. Note: The logic follows what was already there.